Fixup file.line docs to be more clear and consistent

rallytime opened the pull request:
Refs saltstack/salt#31081

Commit:
Fixup file.line docs to be more clear and consistent

Refs #31081

JensRantil commented on salt/modules/file.py:
```diff
@@ -1411,45 +1411,62 @@ def line(path, content, match=None, mode=None, location=None,
+    options.
 
-    :param path:
+    .. note::
```
Would it make sense to add this note after the `mode` argument documentation?

rallytime replied:
Yeah that probably makes sense. I wasn't sure if that should have been exposed up higher in the docs or not. I'll move it. 👍

JensRantil reviewed (commented):
Awesome! Would it make sense to also document the `file.line` state in this PR?

Commit:
Move note about using mode=insert with location options to mode section

Commit:
Clarify file.line state docs as well

rallytime commented:
@JensRantil Thanks for the review! I have moved the note as you suggested for the mode docs and I also added the doc changes to the file.line state function as well. Good catch! 

Do you know if there are any other open issues about the state docs? I will look around, but I thought you might have already filed one that this might close up also. :)

JensRantil commented:
> Do you know if there are any other open issues about the state docs?

Don't think I ever did, but I'm happy you also updated it because it's been equally fuzzy.

JensRantil commented on salt/states/file.py:
```diff
@@ -2653,43 +2653,57 @@ def line(name, content, match=None, mode=None, location=None,
             Place the content at the end of the file.
 
-    :param before:
+    before
```
Is this used for all modes?

rallytime replied:
Good question - it looks like those options only take effect when using either the `insert` or `ensure` modes. I'll add a note about that.

JensRantil commented on salt/states/file.py:
```diff
@@ -2653,43 +2653,57 @@ def line(name, content, match=None, mode=None, location=None,
         Regular expression or an exact case-sensitive fragment of the string.
 
-    :param after:
+    after
```
Is this used for all modes?

rallytime replied:
The modes for `after` are the same as the `before` modes. I'll add a note about this as well.

JensRantil commented on salt/modules/file.py:
```diff
@@ -1411,45 +1411,62 @@ def line(path, content, match=None, mode=None, location=None,
             Place the content at the end of the file.
 
-    :param before:
+    before
```
Is this used for all modes?

rallytime replied:
Same answer here as above: only the ensure and insert modes use this option.

JensRantil commented on salt/modules/file.py:
```diff
@@ -1411,45 +1411,62 @@ def line(path, content, match=None, mode=None, location=None,
         Regular expression or an exact case-sensitive fragment of the string.
 
-    :param after:
+    after
```
Is this used for all modes?

JensRantil commented on salt/modules/file.py:
```diff
@@ -1458,31 +1475,33 @@ def line(path, content, match=None, mode=None, location=None,
         tried to be edited does not exist and nothing really happened.
 
-    :param indent:
+    indent
```
Which modes does this apply to?

JensRantil commented on salt/states/file.py:
```diff
@@ -2698,15 +2712,15 @@ def line(name, content, match=None, mode=None, location=None,
         tried to be edited does not exist and nothing really happened.
 
-    :param indent:
+    indent
```
Which modes does this apply to?

JensRantil reviewed (commented):
Had some more questions that might need clarification.

Commit:
Clarify which modes use "before", "after", and "indent" options

This clarification was added both the the file.line module and
file.line state docs.

rallytime commented:
Thanks again for the good questions @JensRantil. I have added a couple of clarifying sentences to these options for both the module and state docs.
